integrations/spark/spark-job-dataquality/spark-job-dataquality-app/src/main/java/com/thinkbiganalytics/spark/dataquality/output/DataQualityWriter.java

```diff
@@ -0,0 +1,190 @@
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Class to write data quality result to Hive table using SparkSQL <br>
+ * 
+ * This will store the content, convert it to a DataSet and write to Hive. <br>
+ * If the table does not already exists, it will create it.
+ */
+@SuppressWarnings("serial")
+public class DataQualityWriter implements Serializable {
+    private static final Logger log = LoggerFactory.getLogger(DataQualityWriter.class);
+
+    private static final List<DataQualityRow> dqRows = new ArrayList<>();
+    private static DataQualityWriter dqWriter = null;
+    private JavaSparkContext sc = null;
+    private HiveContext hiveContext = null;
+
+    /* no direct instantiation */
+    private DataQualityWriter() {}
+
+    /**
+     * Get the instance of DataQualityWriter
+     *
+     * @return DataQualityWriter
+     */
+    public static DataQualityWriter getInstance() {
+        if (dqWriter == null) {
+            dqWriter = new DataQualityWriter();
+        }
+        return dqWriter;
+    }
+
+    /**
+     * Get list of rows to write
+     *
+     * @return list of DataQualityRow
+     */
+    public List<DataQualityRow> getDataQualityRows() {
+        return dqRows;
+    }
+
+
+    /**
+     * Get count of output rows
+     *
+     * @return row count
+     */
+    public int getDataQualityRowCount() {
+        return dqRows.size();
+    }
+
+    /**
+     * Set Spark context
+     *
+     * @param sc JavaSparkContext
+     */
+    public void setSparkContext(JavaSparkContext sc) {
+        this.sc = sc;
+    }
+
+    /**
+     * Set Hive context
+     *
+     * @param hiveContext HiveContext
+     */
+    public void setHiveContext(HiveContext hiveContext) {
+        this.hiveContext = hiveContext;
+    }
+
+    /**
+     * Add row to write in output
+     *
+     * @param row row for output
+     */
+    public void addRow(DataQualityRow row) {
+        dqRows.add(row);
+    }
+
+    /**
+     * Add multiple rows to write in output
+     *
+     * @param rows list of rows for output
+     */
+    public void addRows(List<DataQualityRow> rows) {
+        dqRows.addAll(rows);
+    }
+
+    /**
+     * Write result to Hive table
+     *
+     * @param scs SparkContextService
+     * @param outputTable Name of the destination Hive table
+     * @param partitionValue Value of the partition for Hive. Currently the processing_dttm column
+     * @return boolean indicating result of write
+     */
+    public boolean writeResultToTable(SparkContextService scs, String outputTable, String partitionValue) {
+
+        boolean retVal = false;
+
+        if (sc == null) {
+            log.error("Error writing result: Spark context is not available.");
+        } else if (hiveContext == null) {
+            log.error("Error writing result: Hive context is not available.");
+        } else if (dqRows.isEmpty()) {
+            log.error("Error writing results: No data to write");
```

Comment:
This can occur if no rows were ingested. Should change to INFO level, with change in message string.